### PR TITLE
build: add devcontainer configuration

### PR DESCRIPTION
**Pull request by lucacasonato**
This allows users to use Codespaces and VS Code Remote Containers easially.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM mcr.microsoft.com/vscode/devcontainers/rust:0-1
+
+ENV DENO_INSTALL=/usr/local
+RUN curl -fsSL https://deno.land/x/install/install.sh | sh
```
Comment (kitsonk) [inline .devcontainer/Dockerfile:4]: missing a return?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      "**/target/**": true
+    }
+  },
+
+  "extensions": [
+    "matklad.rust-analyzer",
+    "bungcip.better-toml",
+    "vadimcn.vscode-lldb",
+    "mutantdino.resourcemonitor",
+    "denoland.vscode-deno"
```
Comment (kitsonk) [inline .devcontainer/devcontainer.json:21]: Do you really want this one installed?  It becomes a little problematic of having a version built in the container? Also, it doesn't do so well with internal code, I have `"deno.enable": false` for my vscode for Deno.  Maybe consider at least setting up some default `deno.` settings in the `devcontainer.json`?

**Review by kitsonk — APPROVED** [review]
Couple thoughts, but awesome.

**Comment by kitsonk** [comment]
The other thing to note, which I haven't figured out, is that this will install Rust 1.53 at the moment.  I haven't figure out how to get Rust 1.54.

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (lucacasonato) [reply .devcontainer/devcontainer.json:21]: Removed.

**Comment by lucacasonato** [comment]
@kitsonk I set it up to install Rust 1.54 through `rustup` now.

**Pull request merged**